### ISSUE TITLE
Project: Move hosting off of Netlify

### ISSUE THREAD

**Issue by marcelgerber** [owid/covid-19-data#2819]
Part of https://github.com/owid/owid-grapher/issues/2751.

With our Netlify contract ending in a few days, we need to move the traffic-heavy covid.ourworldindata.org to some other hosting provider.

Things to watch out for:
- Taking care of the [`_redirects`](https://github.com/owid/covid-19-data/blob/4c721145c89f2017310e205d48eca3b8abfa777d/public/_redirects) and [`_headers`](https://github.com/owid/covid-19-data/blob/4c721145c89f2017310e205d48eca3b8abfa777d/public/_headers_) files
- `Access-Control-Allow-Origin: *` headers
- Currently Netlify is using code-200-redirects to DigitalOcean Spaces - i.e. proxying these requests itself and setting its own headers. Not sure if that is required, probably depends a bit on the `Access-Control-Allow-Origin` situation on DO Spaces.

**Comment by marcelgerber** [comment]
I tried moving it to CF Pages (which would be the easiest option given the existing `_redirects` file and such), but we cannot do that since there are several files that are over the 25MB limit.

Instead, I then tried uploading these files to R2 (bucket `covid-19`) and made that bucket publicly available at https://covid.owid.io.
In addition, I set up _Bulk redirects_ and _Response header transform rules_ in the Cloudflare dashboard, to match `_headers` and `_redirects` as closely as possible.
That seems to have worked pretty well.

There are a few slight differences:
- I couldn't replicate the code-200 redirects from Netlify, so instead these are 301s now.
- I checked, and both CF and DO return `Access-Control-Allow-Origin: *` headers, but _only_ if an `Origin` header is present on the header.

Outstanding steps:
- [ ] Test that explorer files load just fine from covid.owid.io
- [ ] Copy these bulk redirects and transform rules over to apply to covid.ourworldindata.org
- [ ] Update the existing on-deploy workflow to instead upload these files to R2.
- [ ] Switch covid.ourworldindata.org over to point to R2.
